theano/gof/utils.py

```diff
@@ -210,6 +213,15 @@ def __str__(self):
 
         return type.__new__(cls, name, bases, dct)
 
+    _cache = {}
+
+    def __call__(cls, *args, **kwargs):
+        if not cls._has_props:
```

Comment:
I earlier opened a PR, where I made all the `OP` instance have `__props__`. In that, I had left out `ScalarOp` and @abergeron explains here why,  https://github.com/Theano/Theano/pull/4749#issuecomment-252945665 . Because of that, I couldn't proceed with the PR that addresses a centralised caching approach. Also, from what I remember this would fail with the old backend as they don't have `__props__`.